### ISSUE TITLE
'word_size' type issue and  dataset object miss issue

### ISSUE THREAD

**Issue by kunshen1** [intel-analytics/BigDL#9625]
**Type error for 'word size**
```
Traceback (most recent call last):
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 357, in <module>
    fire.Fire(train)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 157, in train
    gradient_accumulation_steps = gradient_accumulation_steps // world_size

```

**CMD: python ./alpaca_qlora_finetuning_cpu.py     --base_model /home/Llama-2-7b-hf/     --data_path /home/dataset/gpt2-peft-abirate-english-quotes/     --output_dir ./bigdl-qlora-alpaca**

ERROR
```
Traceback (most recent call last):
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 357, in <module>
    fire.Fire(train)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 261, in train
    model = get_peft_model(model, config)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/bigdl/llm/transformers/qlora.py", line 170, in get_peft_model
    model = get_peft_model_original(*args, **kwargs)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/mapping.py", line 106, in get_peft_model
    return MODEL_TYPE_TO_PEFT_MODEL_MAPPING[peft_config.task_type](model, peft_config, adapter_name=adapter_name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/peft_model.py", line 889, in __init__
    super().__init__(model, peft_config, adapter_name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/peft_model.py", line 111, in __init__
    self.base_model = PEFT_TYPE_TO_MODEL_MAPPING[peft_config.peft_type](
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/tuners/lora.py", line 274, in __init__
    super().__init__(model, config, adapter_name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/tuners/tuners_utils.py", line 88, in __init__
    self.inject_adapter(self.model, adapter_name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/tuners/tuners_utils.py", line 219, in inject_adapter
    self._create_and_replace(peft_config, adapter_name, target, target_name, parent, **optionnal_kwargs)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/peft/tuners/lora.py", line 372, in _create_and_replace
    new_module = self._create_new_module(lora_config, adapter_name, target, **kwargs)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/bigdl/llm/transformers/qlora.py", line 139, in _create_new_module
    "qa_lora": lora_config.qa_lora,
AttributeError: 'LoraConfig' object has no attribute 'qa_lora'
```

**Comment by Uxito-Ada** [comment]
Hi, this is related to the recent PR #9551 which adds `qa_lora` property. You can walk around it by replacing:
```python
from peft import LoraConfig
```
with
```python
from bigdl.llm.transformers.qlora import LoraConfig
```

We are also updating CPU example of alpaca qlora, and you can wait for us as well

**Comment by kunshen1** [comment]
Thanks, I replace the import, it passed but occur another issue
```
Traceback (most recent call last):
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 358, in <module>
    fire.Fire(train)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/root/miniconda3/envs/llm/lib/python3.9/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/BigDL/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/./alpaca_qlora_finetuning_cpu.py", line 307, in train
    args = transformers.TrainingArguments(
TypeError: __init__() got an unexpected keyword argument 'gradient_checkpointing_kwargs'
```

**Comment by Uxito-Ada** [comment]
Fixed in #9631

**Comment by kunshen1** [comment]
For script fine-tune script still remains "word_size" type issue

ERROR:
![image](https://github.com/intel-analytics/BigDL/assets/54119762/406926ee-71e6-4b18-990b-c38b99a1a9c4)

**Comment by Uxito-Ada** [comment]
> For fine-tune script still remains "word_size" type issue
> 
> ERROR: ![image](https://private-user-images.githubusercontent.com/54119762/290056752-406926ee-71e6-4b18-990b-c38b99a1a9c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0MzUxNjIsIm5iZiI6MTcwMjQzNDg2MiwicGF0aCI6Ii81NDExOTc2Mi8yOTAwNTY3NTItNDA2OTI2ZWUtNzFlNi00YjE4LTk5MGItYzM4Yjk5YTFhOWM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDAyMzQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYWYxYThlMGM4MDEwYjVjZWVhZWMwNGFlOTM4YjAzMjlkNWU5ODM4ZDQ2N2M4OWU4M2FhMTAyMWU2ZDMzYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.criDTRj0R59kSz--7eOqtfzuQWzANajkrglzz3eA9mg)

Hi @hzjane , pls look at this. It seems `get_int_from_env` here returned a str rather than expected int: https://github.com/intel-analytics/BigDL/blob/main/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/alpaca_qlora_finetuning_cpu.py#L145

**Comment by Uxito-Ada** [comment]
> > For fine-tune script still remains "word_size" type issue
> > ERROR: ![image](https://private-user-images.githubusercontent.com/54119762/290056752-406926ee-71e6-4b18-990b-c38b99a1a9c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0MzUxNjIsIm5iZiI6MTcwMjQzNDg2MiwicGF0aCI6Ii81NDExOTc2Mi8yOTAwNTY3NTItNDA2OTI2ZWUtNzFlNi00YjE4LTk5MGItYzM4Yjk5YTFhOWM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDAyMzQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYWYxYThlMGM4MDEwYjVjZWVhZWMwNGFlOTM4YjAzMjlkNWU5ODM4ZDQ2N2M4OWU4M2FhMTAyMWU2ZDMzYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.criDTRj0R59kSz--7eOqtfzuQWzANajkrglzz3eA9mg)
> 
> Hi @hzjane , pls look at this. It seems `get_int_from_env` here returned a str rather than expected int: https://github.com/intel-analytics/BigDL/blob/main/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/alpaca_qlora_finetuning_cpu.py#L145

Has been fixed in #9667 , @kunshen1 you can try it.

**Comment by kunshen1** [comment]
Great, thx!
